### PR TITLE
test/libmpv_encode: explicitly discard fread result

### PR DESCRIPTION
**Pull request by llyyr**
Fixes warning with clang

Read this before you submit this pull request:
https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md

Reading this link and following the rules will get your pull request reviewed
and merged faster. Nobody wants lazy pull requests.

### CODE REVIEW

**Comment by Dudemanguy** [comment]
stream_bluray has an unhandled fread as well if you want to handle that too.

**Comment by llyyr** [comment]
Done

File: stream/stream_bluray.c [hunk 1]
```diff
@@ -553,7 +553,7 @@ static bool check_bdmv(const char *path)
 
     char data[50] = {0};
 
-    fread(data, 50, 1, temp);
+    (void)fread(data, 50, 1, temp);
```
Comment (kasper93) [inline stream/stream_bluray.c:556]: I would prefer to actually check it, same as `stream_dvdnav` does. https://github.com/mpv-player/mpv/blob/6c59b0272b93e2ed4fb3842a0135e1b5e2c907f3/stream/stream_dvdnav.c#L182

File: test/libmpv_encode.c [hunk 2]
```diff
@@ -96,7 +96,7 @@ static void check_output(FILE *fp)
 
     char magic[4] = {0};
     fseek(fp, 0, SEEK_SET);
-    fread(magic, sizeof(magic), 1, fp);
+    (void)fread(magic, sizeof(magic), 1, fp);
```
Comment (kasper93) [inline test/libmpv_encode.c:99]: Here I don't really care, but maybe instead adding workarounds for warning, just check the return value?

I feel like this convey the intent better.
```diff
diff --git a/test/libmpv_encode.c b/test/libmpv_encode.c
index 51a28bbebe..a2dec9e7d8 100644
--- a/test/libmpv_encode.c
+++ b/test/libmpv_encode.c
@@ -94,11 +94,11 @@ static void check_output(FILE *fp)
     if (size < 100)
         fail("did not encode anything\n");

-    char magic[4] = {0};
+    char magic[4];
     fseek(fp, 0, SEEK_SET);
-    fread(magic, sizeof(magic), 1, fp);
+    size_t ret = fread(magic, sizeof(magic), 1, fp);
     static const char ebml_magic[] = {26, 69, 223, 163};
-    if (memcmp(magic, ebml_magic, 4) != 0)
+    if (ret == 1 && memcmp(magic, ebml_magic, sizeof(magic)) != 0)
         fail("output was not Matroska\n");

     puts("output file ok");
```

File: stream/stream_bluray.c [hunk 3]
```diff
@@ -552,15 +552,17 @@ static bool check_bdmv(const char *path)
         return false;
 
     char data[50] = {0};
```
Comment (kasper93) [inline stream/stream_bluray.c:554]: We can skip the init now.
```suggestion
    char data[50];
```

**Review by kasper93 — APPROVED** [review]

**Pull request merged**